app/controllers/rooms_controller.rb

```diff
@@ -0,0 +1,23 @@
+class RoomsController < ApplicationController
+  before_action :authenticate_user!
+
+  def new
+    @room = Room.new
+  end
+
+  def create
+    @room = Room.new(room_params)
+
+    if @room.save
+      render json: @room, status: :created, location: @room
+    else
+      render json: @room.errors, status: :unprocessable_entity
+    end
+  end
+
+  private
+
+  def room_params
+    params.require(:room).permit(:name, :number_of_beds, :price, :description, :picture, :room_type_id, :user_id)
+  end
+end
```

Comment:
- [ _Optional_ ]:  The code is working as expected, but, make sure that the `user_id` is pre-filled-in in the `create` method or inside the private method `room_params` so that the end user can not tamper with the entered data.

For example, you can use the ActionController Parameters `with_defaults()` method to clean up your params as shown in the bellow code snippet.

```Ruby
def room_params
    params.require(:room).permit(:name, :number_of_beds, :price, :description, :picture, :room_type_id, :user_id).with_defaults(user_id: current_user.id)
 end
```